website/server/controllers/api-v3/tasks.js

```diff
@@ -550,139 +547,16 @@ api.scoreTask = {
-      task.group.approval.requestedDate = new Date();
-
-      let fields = requiredGroupFields.concat(' managers');
-      let group = await Group.getGroup({user, groupId: task.group.id, fields});
-
-      // @TODO: we can use the User.pushNotification function because we need to ensure notifications are translated
-      let managerIds = Object.keys(group.managers);
-      managerIds.push(group.leader);
-      let managers = await User.find({_id: managerIds}, 'notifications preferences').exec(); // Use this method so we can get access to notifications
-
-      let managerPromises = [];
-      managers.forEach((manager) => {
-        manager.addNotification('GROUP_TASK_APPROVAL', {
-          message: res.t('userHasRequestedTaskApproval', {
-            user: user.profile.name,
-            taskName: task.text,
-          }, manager.preferences.language),
-          groupId: group._id,
-          taskId: task._id, // user task id, used to match the notification when the task is approved
-          userId: user._id,
-          groupTaskId: task.group.taskId, // the original task id
-          direction,
-        });
-        managerPromises.push(manager.save());
-      });
-
-      managerPromises.push(task.save());
-      await Promise.all(managerPromises);
-
-      throw new NotAuthorized(res.t('taskApprovalHasBeenRequested'));
-    }
-
-    let wasCompleted = task.completed;
-
-    let [delta] = common.ops.scoreTask({task, user, direction}, req);
-    // Drop system (don't run on the client, as it would only be discarded since ops are sent to the API, not the results)
-    if (direction === 'up') common.fns.randomDrop(user, {task, delta}, req, res.analytics);
-
-    // If a todo was completed or uncompleted move it in or out of the user.tasksOrder.todos list
-    // TODO move to common code?
-    let taskOrderPromise;
-    if (task.type === 'todo') {
-      if (!wasCompleted && task.completed) {
-        // @TODO: mongoose's push and pull should be atomic and help with
-        // our concurrency issues. If not, we need to use this update $pull and $push
-        taskOrderPromise = user.update({
-          $pull: { 'tasksOrder.todos': task._id },
-        }).exec();
-        // user.tasksOrder.todos.pull(task._id);
-      } else if (wasCompleted && !task.completed && user.tasksOrder.todos.indexOf(task._id) === -1) {
-        taskOrderPromise = user.update({
-          $push: { 'tasksOrder.todos': task._id },
-        }).exec();
-        // user.tasksOrder.todos.push(task._id);
-      }
-    }
-
-    setNextDue(task, user);
-
-    let promises = [
-      user.save(),
-      task.save(),
-    ];
-
-    if (task.group && task.group.taskId) {
-      await handleSharedCompletion(task);
-    }
-
-    // Save results and handle request
-    if (taskOrderPromise) promises.push(taskOrderPromise);
-    let results = await Promise.all(promises);
+    const data = await scoreTasks(user, [{id: taskId, direction}], req, res);
+    user = data.user;
```

Comment:
I don't think this is necessary, if the context of the user object isn't lost (and it shouldn't) at this point the previous `user` value will have the updated data